docs/guide.adoc

```diff
@@ -745,51 +716,50 @@ Spider :: blob {
     eat_bug: fn -> void,
 }
 
-new_spider :: fn x: float, y: float -> Spider do
-    self: !Spider = nil // Set up the variable
-    self = Spider {
+new_spider :: fn x, y ->
+    Spider {
         hp: 5,
         position: position,
         eat_bug: fn do
             self.hp += 1
         end,
     }
-    ret self
 end
 
 spider := new_spider(0.0, 0.0)
 ----
 
-Notice that the `Spider` blob has many fields in common with the `Creature`
-blob from before. This fact can be used to do something called
-https://en.wikipedia.org/wiki/Duck_typing[duck typing], which effectively
-allows us to assign `Spider` blobs to `Creature` variables.
-
-.Duck typing
-[source, sylt]
-----
-creatures: [Creature] = []
-creatures -> push' new_spider' 0.0, 0.0
-creatures[0].hp -= 1   // Ok
-creatures[0].eat_bug() // Type error
-----
-
-This type error may seem counter-intuitive. We know that `creatures[0]` is a
-`Spider` which has the `eat_bug` function, yet we are not allowed to call it.
-What is really going on here? Once the `Spider` blob is put into the list
-of type `[Creature]` some type information is lost, meaning, there is no
-reliable way of knowing the true type of something in the list. `creatures` is
-a list of `Creature`s and other blobs containing the same fields as `Creature`,
-though every element is treated as if being of type `Creature`.
-
-To call the `eat_bug` function anyway, forced types may be used.
-
-.Force general blob into specific blob
-[source, sylt]
-----
-spider: !Spider = creatures[0]
-spider.eat_bug() // Ok, as long as creatures[0] is a Spider
-----
+// Leaving this here until we decide what happens with ducktyping
+//Notice that the `Spider` blob has many fields in common with the `Creature`
+//blob from before. This fact can be used to do something called
+//https://en.wikipedia.org/wiki/Duck_typing[duck typing], which effectively
+//allows us to assign `Spider` blobs to `Creature` variables.
+//
+//.Duck typing
+//[source, sylt]
+//----
+//creatures: [Creature] = []
+//creatures -> push' new_spider' 0.0, 0.0
+//creatures[0].hp -= 1   // Ok
+//creatures[0].eat_bug() // Type error
+//----
+//
+//This type error may seem counter-intuitive. We know that `creatures[0]` is a
+//`Spider` which has the `eat_bug` function, yet we are not allowed to call it.
+//What is really going on here? Once the `Spider` blob is put into the list
+//of type `[Creature]` some type information is lost, meaning, there is no
+//reliable way of knowing the true type of something in the list. `creatures` is
+//a list of `Creature`s and other blobs containing the same fields as `Creature`,
+//though every element is treated as if being of type `Creature`.
+//
+//To call the `eat_bug` function anyway, forced types may be used.
+//
+//.Force general blob into specific blob
+//[source, sylt]
+//----
+//spider: !Spider = creatures[0]
+//spider.eat_bug() // Ok, as long as creatures[0] is a Spider
+//----
```

Comment:
```suggestion
```
Ducktyping is dead as far as I'm concerned. There's no way to add it into the type-system as it is.